code/globalincs/pool.h

```diff
@@ -0,0 +1,404 @@
+#pragma once
+
+#include "globalincs/pstypes.h"
+#include <tl/optional.hpp>
+#include <csignal>
+//#include <vector>
+
+using  tl::optional;
+struct pool_index{
+	private:
+	int index;
+	int generation;
+	bool is_null = true;
+	public:
+	pool_index(){ nullify();};
+	pool_index(int i,int g,bool b = false){
+		if(b){ nullify();
+		}
+		else{
+			set(i,g);
+		}
+	};
+	void set(int i,int g){
+		is_null = false;
+		index = i;
+		generation = g;
+	}
+	void nullify()
+	{
+		index = -1;
+		generation = -1;
+		is_null=true;
+	};
+	bool operator==(pool_index &rhs) const {
+		if(is_null && rhs.is_null)
+			return true;
+		if(is_null || rhs.is_null)
+			return false;
+		return (index == rhs.index && generation == rhs.generation);
+	};
+	pool_index& operator=(pool_index const&rhs) {
+		if(rhs.is_null)
+			nullify();
+		else
+			set(rhs.index, rhs.generation);
+		return *this;
+	};
+	int i() const { return index;};
+	int g() const { return generation;};
+	bool null() const { return is_null;};
+	bool has_value() const { return !is_null;};
```

Comment:
Given that `inline` is, at best, interpreted as a weak suggestion by compilers, this seems unnecessary to me

Reply:
Documentation I found about inline also suggests that these would be treated as marked by it anyway... but I put it on anyway as it might be useful as a hint to future programmers.